src/ORM/DataList.php

```diff
@@ -1252,4 +1252,27 @@ public function offsetUnset($key)
     {
         user_error("Can't alter items in a DataList using array-access", E_USER_ERROR);
     }
+
+    /**
+     * Iterate over this datalist in "chunks". This will break the query in smaller subset and avoid loading the entire
+     * result set in memory at once. Beware not to perform any operations on the results that might alter the return
+     * order. Otherwise, you might break subsequent chunks.
+     * @param int $chunkSize
+     * @return Generator|DataObject[]
+     */
+    public function chunk($chunkSize = 100)
+    {
+        if ($chunkSize < 1) {
+            throw new InvalidArgumentException('DataList::chunk: chunksize must be greater than or equal to 1');
```

Comment:
Can you add a PHPDoc for this?

Reply:
You could also use `__METHOD__` in the message here.

Reply:
Added a `@throws` statement and switched to `__CLASS__` and `__METHOD__` for the exception message.